dialog/color.go

```diff
@@ -0,0 +1,334 @@
+package dialog
+
+import (
+	"fmt"
+	"image/color"
+	"math"
+	"strings"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// ColorPickerDialog is a simple dialog window that displays a color picker.
+type ColorPickerDialog struct {
+	*dialog
+	color    color.Color
+	callback func(c color.Color)
+	advanced *widget.AccordionContainer
+	picker   *colorAdvancedPicker
+}
+
+// NewColorPicker creates a color dialog and returns the handle.
+// Using the returned type you should call Show() and then set its color through SetColor().
+// The callback is triggered when the user selects a color.
+func NewColorPicker(title, message string, callback func(c color.Color), parent fyne.Window) *ColorPickerDialog {
+	p := &ColorPickerDialog{
+		dialog:   newDialog(title, message, theme.ColorPaletteIcon(), nil /*cancel?*/, parent),
+		color:    theme.PrimaryColor(),
+		callback: callback,
+	}
+
+	p.picker = newColorAdvancedPicker(theme.PrimaryColor(), func(c color.Color) {
+		p.color = c
+	})
+	p.advanced = widget.NewAccordionContainer(widget.NewAccordionItem("Advanced", p.picker))
+
+	p.dialog.content = widget.NewVBox(fyne.NewContainerWithLayout(layout.NewCenterLayout(), widget.NewVBox(p.createSimplePickers()...)), p.advanced)
+
+	p.dialog.dismiss = &widget.Button{Text: "Cancel", Icon: theme.CancelIcon(),
+		OnTapped: p.dialog.Hide,
+	}
+	confirm := &widget.Button{Text: "Confirm", Icon: theme.ConfirmIcon(), Style: widget.PrimaryButton,
+		OnTapped: func() {
+			p.selectColor(p.color)
+		},
+	}
+	p.dialog.setButtons(newButtonList(p.dialog.dismiss, confirm))
+	return p
+}
+
+// NewSimpleColorPicker creates a simple color dialog and returns the handle.
+// Using the returned type you should call Show() and then set its color through SetColor().
+// The callback is triggered when the user selects a color.
+func NewSimpleColorPicker(title, message string, callback func(c color.Color), parent fyne.Window) *ColorPickerDialog {
```

Comment:
I'm not really seeing the benefit of two APIs here (SimpleColorPicker vs ColorPicker) - for one thing the docs don't even explain the difference.
You are returning a ColorPickerDialog, so I think that "ShowAdvanced" or "HideAdvanced" or some other bool would be appropriate to expose the fact that developers can decide whether or not that option is available.
So `p := dialog.NewColorPicker(); p.HideAdvanced; p.Show()` would remove the need for the simple api variants.